astropy/io/ascii/latex.py

```diff
@@ -112,7 +112,7 @@ def process_line(self, line):
         line = RE_COMMENT.split(line)[0]
         line = line.strip()
         if line.endswith(r"\\"):
-            line = line.rstrip(r"\\")
```

Comment:
`rstrip` does not raise an error either, so that may just have been an optimisation attempt, but I agree the one-liner is even neater.

Reply:
It can almost be refactored like this, but the code manually raises an error if the suffix is missing, so the best we can do is 
```python
        line = RE_COMMENT.split(line)[0].strip()
        if line.endswith(r"\\"):
            line = line.removesuffix(r"\\")
        else:
            raise core.InconsistentTableError(
                r"Lines in LaTeX table have to end with \\"
            )
```

Reply:
I should have read more of the code before making my first suggestion, but having done so now I would suggest the following:
```python
        line = RE_COMMENT.split(line)[0].strip()
        if not line.endswith(r"\\"):
            raise core.InconsistentTableError(
                r"Lines in LaTeX table have to end with \\"
            )
        return line.removesuffix(r"\\")
```

Reply:
SGTM.

Reply:
Done.